lib/helper.js

```diff
@@ -79,18 +79,17 @@ function stubComponent(tag, children, showDataProps) {
     return React.createClass({
         displayName: tag.displayName || tag,
 
-        componentWillMount: function () {
-            showDataProps && this.addDataProps();
-        },
+        render: function () {
+            var props = {};
 
-        addDataProps: function () {
             Object.keys(this.props).forEach(function (key) {
-                this.props['data-' + key] = this.props[key]
+                props[key] = this.props[key];
+                if (showDataProps) {
+                    props['data-' + key.toLowerCase()] = this.props[key];
+                }
             }, this);
-        },
 
-        render: function () {
-            return React.createElement(tag, this.props, children);
+            return React.createElement(tag, props, children);
```

Comment:
@longlho can you move the logic to set the pros to an other method so that the render method is 1 line:

```js
render: function () {
    return React.createElement(tag, this.getStubProps(), children);
}
```